### PR TITLE
Added emulator deep links to all samples

### PR DESCRIPTION
**Pull request by tonyanziano**
Retry of #587 from my own fork because for some reason git started denying me push access to the base repo.

---

- Added deep link to emulator for all samples except for:
  - **Luis Dispatch** - no `.bot` file
  - **SimplePrompt** - proposed to be dropped due to changed code (@johnataylor)
- Added MVC / Razor pages so that deep links can be generated in C# before being passed to the client-side html

### CODE REVIEW

**Comment by tonyanziano** [comment]
Unable to add reviewers, so: @cleemullins @drub0y

File: samples-final/2.AspNetCore-ConversationUpdate-Bot/AspNetCore-ConversationUpdate-Bot.csproj [hunk 1]
```diff
@@ -6,9 +6,8 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.7" />
-    <PackageReference Include="Microsoft.Bot.Builder.Core" Version="4.0.1-preview" />
-    <PackageReference Include="Microsoft.Bot.Builder.Core.Extensions" Version="4.0.1-preview" />
-    <PackageReference Include="Microsoft.Bot.Builder.Integration.AspNet.Core" Version="4.0.1-preview" />
+    <PackageReference Include="Microsoft.Bot.Builder" Version="4.0.0-local" />
```
[the remark this answers is not in the document]
Comment (johnataylor) [reply samples-final/2.AspNetCore-ConversationUpdate-Bot/AspNetCore-ConversationUpdate-Bot.csproj]: Yes Chris's comment is correct. we want these when checked in to be against the latest released. (The version number there will presumably be something like 4.0.1-preview2.)

File: samples-final/2.AspNetCore-ConversationUpdate-Bot/Startup.cs [hunk 2]
```diff
@@ -41,12 +40,14 @@ public void ConfigureServices(IServiceCollection services)
                 // Any exceptions thrown by other Middleware, or by your OnTurn method, will be 
                 // caught here. To facillitate debugging, the exception is sent out, via Trace, 
                 // to the emulator. Trace activities are NOT displayed to users, so in addition
-                // an "Ooops" message is sent. 
-                options.Middleware.Add(new CatchExceptionMiddleware<Exception>(async (context, exception) =>
+                // an "Ooops" message is sent.
+
+                // TODO: Update block against new code. (CatchExceptionMiddleware doesn't exist)
```
[the remark this answers is not in the document]
Comment (johnataylor) [reply samples-final/2.AspNetCore-ConversationUpdate-Bot/Startup.cs]: If we decide we want the ErrorHandler here across all the samples we can do that as a separate check in. If you want to do that go ahead and open an issue and assign it it me.

**Comment by johnataylor** [comment]
As a separate check-in we will be dropping SimplePrompt.

File: samples-final/5.AspNetCore-MultiplePrompts-Bot/Pages/Index.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        public IndexModel(IHostingEnvironment hostingEnv, IOptions<BotFrameworkOptions> options)
+        {
+            _hostingEnv = hostingEnv;
+            _options = options;
+        }
+
+        public void OnGet()
+        {
+            string botUrl = $"{ Request.Scheme }://{ Request.Host }/api/messages";
```
[the remark this answers is not in the document]
Comment (tonyanziano) [reply samples-final/5.AspNetCore-MultiplePrompts-Bot/Pages/Index.cshtml.cs]: good catch 👍

**Comment by tonyanziano** [comment]
Ok I resolved the errors previously mentioned in the **Luis-Dispatch** sample, but there is still a reference in that sample to the `LuisRecognizerMiddleware` which is used to get the `dispatchResult` in `OnTurnAsync()`. So should we proceed with this sample in the same manner as the other 2 referring to deleted middleware?

**Comment by cleemullins** [comment]
Closed as no longer relevant. Samples now live in the botbuilder-samples repo.

**Pull request closed without merging**